sway-core/src/parse_tree/expression/method_name.rs

```diff
@@ -8,18 +8,27 @@ pub enum MethodName {
     /// Represents a method lookup with a type somewhere in the path
     FromType {
         call_path: CallPath,
-        type_name: Option<TypeInfo>,
-        type_name_span: Option<Span>,
+        type_name: TypeInfo,
+        type_name_span: Span,
     },
     /// Represents a method lookup that does not contain any types in the path
+    /// something like a.b(c)
+    /// in this case, the first argument defines where to look for the method
     FromModule { method_name: Ident },
+    /// something like a::b::c()
+    /// in this case, the path defines where the fn symbol is defined
+    /// used for things like core::ops::add(a, b).
```

Comment:
From my understanding, it is not possible to call `core::ops::add(a, b)` in user-facing Sway. Can you either add more explanation about how `core::ops::add` is not user-facing code, or provide an example of user-facing Sway code that would trigger this case?

Reply:
👍